### ISSUE TITLE
Error (criu/cr-restore.c:1130): 24 killed by signal 11: Segmentation fault

### ISSUE THREAD

**Issue by avagin** [xemul/criu#304]
$ make ASAN=1
$ export SKIP_PREP=1
$ ./test/jenkins/criu-fcg.sh
```
===================== Run zdtm/transition/thread-bomb in h =====================
Start test
./thread-bomb --pidfile=thread-bomb.pid --outfile=thread-bomb.out
03:56:43.367:    23: 1024
Run criu pre-dump
Run criu pre-dump
Run criu pre-dump
Run criu dump
Run criu restore
=[log]=> dump/zdtm/transition/thread-bomb/24/4/restore.log
------------------------ grep Error ------------------------
(00.270542)     24: 			Going to dup 1 into 2
(00.270548)     24: 	Receive fd for 2
(00.270557)     24: 		Create fd for 3
(00.337290)     24: Restore via sigreturn
(01.122338) Error (criu/cr-restore.c:1130): 24 killed by signal 11: Segmentation fault
(01.131860) Error (criu/cr-restore.c:2059): Restoring FAILED.
------------------------ ERROR OVER ------------------------
############ Test zdtm/transition/thread-bomb FAIL at CRIU restore #############
```

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x0000000000457d77 in rst_prep_creds_args (ce=0x60e0000281a0, prev_pos=prev_pos@entry=0x7ffca58d8de0) at criu/cr-restore.c:2873
2873			creds->uid = target_userns_uid(ns, creds->uid);
Missing separate debuginfos, use: dnf debuginfo-install glibc-2.23.1-11.fc24.x86_64 libasan-6.3.1-1.fc24.x86_64 libgcc-6.3.1-1.fc24.x86_64 libnet-1.1.6-11.fc24.x86_64 libnl3-3.2.28-4.fc24.x86_64 libselinux-2.5-12.fc24.x86_64 libstdc++-6.3.1-1.fc24.x86_64 pcre-8.40-5.fc24.x86_64 protobuf-c-1.2.1-1.fc24.x86_64
(gdb) bt
#0  0x0000000000457d77 in rst_prep_creds_args (ce=0x60e0000281a0, prev_pos=prev_pos@entry=0x7ffca58d8de0) at criu/cr-restore.c:2873
#1  0x0000000000459f46 in rst_prep_creds (core=0x7ffca58d91e0, creds_pos=<synthetic pointer>, pid=0) at criu/cr-restore.c:2954
#2  sigreturn_restore (pid=pid@entry=24, task_args=task_args@entry=0x7f9fae659000, alen=alen@entry=430080, core=core@entry=0x60800000b9a0) at criu/cr-restore.c:3001
#3  0x000000000045e6a7 in restore_one_alive_task (pid=<optimized out>, core=core@entry=0x60800000b9a0) at criu/cr-restore.c:733
#4  0x000000000046221b in restore_one_task (core=0x60800000b9a0, pid=<optimized out>) at criu/cr-restore.c:899
#5  restore_task_with_children (_arg=<optimized out>) at criu/cr-restore.c:1466
#6  0x00007f9fac4360ed in clone () from /lib64/libc.so.6
(gdb) p args->creds 
$5 = {base = {descriptor = 0x56c280 <creds_entry.descriptor>, n_unknown_fields = 0, unknown_fields = 0x0}, uid = 18943, gid = 58467, euid = 18943, egid = 58467, suid = 18943, sgid = 58467, 
  fsuid = 18943, fsgid = 58467, n_cap_inh = 2, cap_inh = 0x0, n_cap_prm = 2, cap_prm = 0x0, n_cap_eff = 2, cap_eff = 0x0, n_cap_bnd = 2, cap_bnd = 0x0, secbits = 0, n_groups = 2, 
  groups = 0x0, lsm_profile = 0x0}
(gdb) p &args->creds 
$6 = (CredsEntry *) 0x7f9fae646f18
(gdb) p creds 
$7 = (CredsEntry *) 0x7f9fab6fff18
```

**Comment by avagin** [comment]
Cc: @0x7f454c46 @cyrillos

**Comment by avagin** [comment]
```
(gdb) fram 1
#1  0x0000000000459f46 in rst_prep_creds (core=0x7ffd270f7d50, creds_pos=<synthetic pointer>, pid=0) at criu/cr-restore.c:2954
2954			args = rst_prep_creds_args(ce, &this_pos);
(gdb) p i
$3 = 456
```

**Comment by xemul** [comment]
Is it purely for dev branch?

**Comment by avagin** [comment]
@xemul yes

**Comment by avagin** [comment]
```
commit a19f8aada1cb77b228f339f5f1be8ad131cbcd7c
Author: Andrei Vagin <avagin@virtuozzo.com>
Date:   Thu Mar 30 06:42:15 2017 +0300

    restore: don't forget to update creds after rst_mem_remap_ptr()
    
    rst_mem_alloc() can moves a vma with previous objects, so
    if we want to access them, we have to update their pointers
    
    https://github.com/xemul/criu/issues/304
    
    Cc: Kirill Tkhai <ktkhai@virtuozzo.com>
    Fixes: 72e295ebbb26 ("ns: Convert task cred's xids to target user ns")
    Signed-off-by: Andrei Vagin <avagin@virtuozzo.com>
    Reviewed-by: Kirill Tkhai <ktkhai@virtuozzo.com>
    Signed-off-by: Andrei Vagin <avagin@virtuozzo.com>
```
